scripts/utils/starknet.py

```diff
@@ -246,12 +283,63 @@ def _convert_offset_to_hex(obj):
     )
 
 
+async def deploy_starknet_account(private_key=None, amount=1) -> Account:
+    compile_contract(
+        {"contract_name": "OpenzeppelinAccount", "is_account_contract": True}
+    )
+    class_hash = await declare("OpenzeppelinAccount")
+    salt = random.randint(0, 2**251)
+    global _private_key
+    global _account_address
+    private_key = private_key or _private_key
+    if private_key is None:
+        raise ValueError(
+            "private_key was not given in arg nor in env variable, see README.md#Deploy"
+        )
+    key_pair = KeyPair.from_private_key(int(private_key, 16))
+    constructor_calldata = [key_pair.public_key]
+    address = compute_address(
+        salt=salt,
+        class_hash=class_hash,
+        constructor_calldata=constructor_calldata,
+        deployer_address=0,
+    )
+    logger.info(f"ℹ️  Funding account {hex(address)} with {amount} ETH")
+    await fund_address(address, amount=amount)
+    logger.info(f"ℹ️  Deploying account")
+    res = await Account.deploy_account(
+        address=address,
+        class_hash=class_hash,
+        salt=salt,
+        key_pair=key_pair,
+        client=RPC_CLIENT,
+        chain=CHAIN_ID,
+        constructor_calldata=constructor_calldata,
+        max_fee=int(1e17),
+    )
+    status = await wait_for_transaction(res.hash)
+    if status == TransactionStatus.REJECTED:
+        logger.warning("⚠️  Transaction REJECTED")
+
+    logger.info(f"✅ Account deployed at address {hex(res.account.address)}")
+    _account_address = hex(res.account.address)
+    _private_key = hex(key_pair.private_key)
+    return res.account
+
+
 async def declare(contract_name):
     logger.info(f"ℹ️  Declaring {contract_name}")
-    account = await get_starknet_account()
     artifact = get_artifact(contract_name)
     compiled_contract = Path(artifact).read_text()
     contract_class = create_compiled_contract(compiled_contract=compiled_contract)
+    class_hash = compute_class_hash(contract_class=deepcopy(contract_class))
+    try:
+        await RPC_CLIENT.get_class_by_hash(class_hash)
```

Comment:
it is in this moment that I miss Result<> type to handle this, but anyway^^
i'm starting to hate the try catch style of javascript and python